### DISCUSSION TITLE
Different behaviour  between Z80Pack Cromemcosim and Cromemco Z-1 Replica - Cromix HARD DISK BOOT

### DISCUSSION THREAD

**Discussion by riwide** [thehighnibble/thehighnibble.github.io#58] [Cromemco Z-1 replica]
I can boot HD0.HDD (Cromix) from Cromemcosim , but if I transfer the file across to Z-1 Replica and try to boot it, then I get an error

Floppy = 1, Hard disk = 2
Enter major root device number: 2
hd0 = 0, hd1 = 1, hd2 = 2
Enter minor root device number: 0
Unable to read super-block

Why would there be a difference - is  not the Z-1 replica using functionally similar software to Z80pack?

**Comment by udo-munk** [comment]
You must configure the HD in boot.conf, have you done this?

**Comment by riwide** [comment]
yes, in the way that the z1 says, and from a floppy boot, the z1 hd0 can be
worked on with newdisk.cmd. I am not experienced enough yet to mount the
z80pack hd0  on z1 after a floppy boot, but I believe it is configured
correctly.
Surely if boot.conf was not correct the system would not progress to the
point of giving superblock not found?

I have been happily accessing both floppy and hard disk from cdos boot,
using drive letters, so I assume I am doing the correct config in boot.conf.

on Z1 I have:

A:DSK: = "udo1127.dsk"

B:DSK: = "cdos258_8.dsk"

C:DSK: = ""

D:DSK: = ""

HD0:[C8-1] = "old_hd0.hdd"

HD1: OFFLINE

HD2: OFFLINE

udo1127.dsk is probably same as Z1  cromix1127_8.dsk, but I was checking to
see if I could find hdinit.

I found that Z1 latest image cromix disk did not have the hdinit and
possible hd support that is described in THN #037 video, so I tried to
bring a working bootable hd from cromemcosim across to Z1.

All good fun, but happy to find a solution :)

Richard

**Comment by udo-munk** [comment]
I do not have a Z1 replica, so I can test only with z80pack cromemcosim. If I try to boot Cromix from HD and there is no HD image file available, or wrong filename I get:

Floppy = 1, Hard disk = 2
Enter major root device number: 2
hd0 = 0, hd1 = 1, hd2 = 2
Enter minor root device number: 0

Disk home error: Dev 2:0, Blk -1, Cyl 00, Surf 0, Sec 00, Stat 700
Unable to read super-block

Floppy = 1, Hard disk = 2
Enter major root device number:

**Comment by riwide** [comment]
Thanks,  I will check further. I will rename to hd0.hdd and try again,
since that is "more" of a default. It it still fails I shall have to ask
Davd McNaughton to try.

Cheers
Richard

**Comment by pfcushing** [comment]
Richard, In that video, Dave alluded to the fact that CDOS needed to be reconfigured in order to add harddisk support but did not go into any detail on how to actually do that.  There is a "utility" called CDOSGEN that you run to build a new CDOS258 image on the disk.  You will need to answer several questions about the floppy and harddisks that you want supported.  Reading the CDOS manual ahead of time will benefit you greatly when answering those questions.

Good luck
Paul Cushing

**Comment by riwide** [comment]
Thanks for the tip! I shall try that. I was working towards that so that I
know far more about the Cromemco environment than currently.

Richard

**Comment by riwide** [comment]
I have now been able to generate a CDOS and Cromix  hard disk, and make the Cromix hd bootable. CDOSGEN and CROGEN solved the problems.
